### PR TITLE
feat(recordings): add recording playlist behind ff

### PR DESCRIPTION
**Pull request by rcmarron**
## Problem

Users want to watch lots of recordings at once, but today you have to click back and forth

## Changes

Adds a feature flag `session-recording-playlist` that changes the recording list to a playlist with a player on the side. A couple notes on the implementations:
* All player logics are keyed with a `sessionRecordingId` + a `playerKey`. This isolates things like player states etc
* Settings that should span recordings are put into a new `playerSettingsLogic`
* Renamed `sessionRecordingLogic` to `sessionRecordingDataLogic` because now it only manages the data for a recording (it's keyed with a recording id too)
* Extracts out a recording list filter logic + component (because it would've been duplicated)
* Had to re-work the player drawer and as part of that I consolidated logic for how `sessionRecordingId` is added to the urls
* Simplified the resizing of the player frame
* And more!

Things to do in follow up PRs:
- [ ] Update the filter controls
- [ ] Get the console/event list to expand to the bottom of the playlist
- [ ] Analytics
- [ ] Probably more!

Check out the video below!

https://user-images.githubusercontent.com/4813045/186980844-8e3c79db-7435-434c-bb47-241cdf2e1ae8.mov

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Added tests

### CODE REVIEW

File: frontend/src/loadPostHogJS.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export function loadPostHogJS(): void {
                 api_host: window.JS_POSTHOG_HOST,
                 _capture_metrics: true,
                 rageclick: true,
-                debug: window.JS_POSTHOG_SELF_CAPTURE,
```
Comment (alexkim205) [inline frontend/src/loadPostHogJS.tsx:27]: This would be helpful but perhaps this should be in its own separate PR.

**Review by alexkim205 — CHANGES_REQUESTED** [review]

**Review by alexkim205 — CHANGES_REQUESTED** [review]

File: frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts [hunk 2]
```diff
@@ -56,13 +60,16 @@ export const DEFAULT_ENTITY_FILTERS = {
...
+
 export const sessionRecordingsTableLogic = kea<sessionRecordingsTableLogicType>({
     path: (key) => ['scenes', 'session-recordings', 'sessionRecordingsTableLogic', key],
-    key: (props) => props.key || props.personUUID || 'global',
-    props: {} as {
-        personUUID?: PersonUUID
-        key?: string
-    },
+    props: {} as SessionRecordingTableLogicProps,
+    key: (props) => `${props.key || props.personUUID || 'global'}-${props.isPlaylist ? 'playlist' : 'table'}`,
```
[the remark this answers is not in the document]
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts:72]: will be removed w/ FF's (discussed in meeting)

**Review by alexkim205 — APPROVED** [review]

File: frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts [hunk 3]
```diff
@@ -30,7 +31,10 @@ interface HashParams {
     sessionRecordingId?: SessionRecordingId
 }
 
-const LIMIT = 50
+const TABLE_LIMIT = 50
+const PLAYLIST_LIMIT = 20
```
[the remark this answers is not in the document]
Comment (rcmarron) [reply frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts]: With the playlist design, the recording row is taller, so listing out 50 is really tall and seems odd. And on the flip-side, on the table view, 20 seems short.

File: frontend/src/scenes/session-recordings/sessionPlayerDrawerLogic.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    urlToAction(({ actions, values }) => {
+        const urlToAction = (_: any, __: any, hashParams: HashParams): void => {
+            const nulledSessionRecordingId = hashParams.sessionRecordingId ?? null
+            if (nulledSessionRecordingId !== values.activeSessionRecordingId) {
+                actions.openSessionPlayer(nulledSessionRecordingId, RecordingWatchedSource.Direct)
+            }
+        }
+        return {
+            '*': urlToAction,
+        }
```
[the remark this answers is not in the document]
Comment (rcmarron) [reply frontend/src/scenes/session-recordings/sessionPlayerDrawerLogic.ts:70]: This is on-purpose. We used to have bugs where recordings didn't appear on pages when they should (e.g. the insights shown on the experiments page) because it wasn't obvious that you needed to add a urlToAction. By consolidating this all to the `sessionPlayerDrawer`, we're basically saying, if the `sessionPlayerDrawer` is mounted, then that page should handle recordings in the URL
